File: Semana_2/Cypress/JavaScript/npm/curso/e2e/cypress/pages/separated/login.actions.js
const { LOGIN_LOCATORS } = require('./login.locators');
const { capture } = require('../../support/evidence');

class LoginActions {
  open() {
    cy.visit('/index.htm');
    capture('parabank-separated-open-login');
    return this;
  }

  loginAs(username, password) {
    cy.get(LOGIN_LOCATORS.username).clear().type(username);
    capture('parabank-separated-type-username');
    cy.get(LOGIN_LOCATORS.password).clear().type(password);
    capture('parabank-separated-type-password');
    cy.get(LOGIN_LOCATORS.loginButton).click();
    capture('parabank-separated-click-login');
  }
}

module.exports = { LoginActions };

File: Semana_2/Selenium/Java/maven/curso/e2e/src/test/java/com/curso/selenium/pages/separated/LoginActions.java
package com.curso.selenium.pages.separated;

import com.curso.selenium.evidence.EvidenceSupport;
import org.openqa.selenium.WebDriver;

public class LoginActions {
    private final WebDriver driver;

    public LoginActions(WebDriver driver) {
        this.driver = driver;
    }

    public LoginActions open() {
        driver.get(LoginLocators.URL);
        EvidenceSupport.capture(driver, "parabank-separated-open-login");
        return this;
    }

    public void loginAs(String username, String password) {
        driver.findElement(LoginLocators.USERNAME_INPUT).sendKeys(username);
        EvidenceSupport.capture(driver, "parabank-separated-type-username");
        driver.findElement(LoginLocators.PASSWORD_INPUT).sendKeys(password);
        EvidenceSupport.capture(driver, "parabank-separated-type-password");
        driver.findElement(LoginLocators.LOGIN_BUTTON).click();
        EvidenceSupport.capture(driver, "parabank-separated-click-login");
    }
}
